tests/utils.py

```diff
@@ -37,6 +37,7 @@
 try:
   # is defined in Python 3
   TimeoutError
+  ChildProcessError
```

Comment:
This will not work:
* if TimeoutError fails, you define class ChildProcessError?
* the code that uses this is in another file and uses utils.ChildProcessError -- that isn't even defined in python 3

Reply:
I think it makes sense to just define an error of your own if ChildProcessError is not in python2 (is it's not 100% correct even in python3): `TestServerError` or something.

Reply:
Both `TimeoutError` and `ChildProcessError` are available on python 3.X and not available on python 2.X, so if `TimeoutError` is not found then it makes sense to define both of them.

> the code that uses this is in another file and uses utils.ChildProcessError -- that won't be defined in python 3

for that you are right. I was mistaken in my assumption.

I think that ChildProcessError is a suitable name for that error.
I just think that in `test_utils.py` I will fix that error with:
```
    child_process_error = None
    if sys.version_info.major == 2:
      child_process_error = utils.ChildProcessError
    else:
      child_process_error = ChildProcessError
```

Reply:
> Both `TimeoutError` and `ChildProcessError` are available on python 3.X and not available on python 2.X, so if `TimeoutError` is not found then it makes sense to define both of them.

no. You can either decide things based on python version or you can decide things based on feature testing. Using feature testing as if it is python version testing is wrong even if it happens to work right now.